src/api/app/models/kiwi/image.rb

```diff
@@ -51,6 +53,29 @@ def self.build_from_xml(xml_string, md5)
     end
     new_image
   end
+
+  def to_xml
+    kiwi_file = package.kiwi_image_file
+    return nil unless kiwi_file
+    kiwi_body = package.source_file(kiwi_file)
+
+    xml_repos = repositories.map(&:to_xml).join("\n")
+
+    doc = Nokogiri::XML::DocumentFragment.parse(kiwi_body)
+    doc.xpath("image/repository").remove
+
+    image = doc.at_css('image')
+    image.first_element_child.after(xml_repos)
+
+    # Reparser for pretty printing
+    Nokogiri::XML(doc.to_xml, &:noblanks).to_xml
+  end
+
+  def write_to_backend
+    package.save_file({ filename: package.kiwi_image_file, file: to_xml })
+    self.md5_last_revision = package.kiwi_file_md5
```

Comment:
The check of the kiwi file md5sum is done in the frontend, in the view. Have a look [here](https://github.com/openSUSE/open-build-service/pull/3225/files#diff-030dd5bfde342910fc0161df227de456R15)

As the form button is clicked, a JSON request is trown, and a "is_outdated" state is expected:
- If "is_outdated" is true, a confirm dialog is thrown, asking for the confirmation of the user to apply the changes anyway (different md5sum).
- If "is_outdated" is false, the update action is requested (same md5sum).